### ISSUE TITLE
Numerical continuation and bifurcation toolkit

### ISSUE THREAD

**Issue by balbirthomas** [scipy/scipy#13539]
**The Problem**
Scipy provides many tools to build and simulate mathematical models. However there does not seem to be any means to study dependence of qualitative behavior on model parameters, such as using techniques of numerical continuation and bifurcation analysis. Opens source tools for bifurcation analysis are presently available. A widely used such tool is [Auto](https://github.com/auto-07p) . Another alternative is [bifpack](http://www.bifurcation.de/software.html). Both are written in Fortran. Auto does provide a Python interface but this is a shell like environment to run the simulations. The models themselves still need to be specified in Fortran. Another Python interface to Auto is [PyDSTool](https://pypi.org/project/PyDSTool/). PyDSTool requires the model (system of differential equations) be specified as a string. Auto and PyDSTool make these design choices for reasons of computational efficiency.  

**Ideal Solution**
It would be ideal if it were possible to code and analyze models only using Python. Is this possible ? For instance could code generation be used to speed up computation. A Python function may be used to specify a model, say a system of ordinary differential equations, just as is done for `scipy.integrate.odeint`. The function `f(y, t, ...)` used by `odeint` may depend on a list of parameters `p` (`f(y, t, p, ...)`). Could C/C++ or Fortran code for such a function be generated transparently and passed to numerical routines such as Auto or Bifpack in order to compute the bifurcation diagram. There seems to be at least one package ( [Transpyle](https://pypi.org/project/transpyle/) ) that does transpile Python to Fortran. Is there a better way to achieve the same, such as using [Cython](https://cython.org/) ?

**Similar Tools**
These are some examples of other bifurcation analysis toolkits
- [Bifurcationkit.jl](https://github.com/rveltz/BifurcationKit.jl)
- [Coco](https://sourceforge.net/projects/cocotools/)
- [Matcont](https://sourceforge.net/projects/matcont/)

**Comment by rkern** [comment]
I recommend asking on the [scipy-user mailing list](https://www.scipy.org/scipylib/mailing-lists.html). Github issues are not productive places for conversational questions like this.

**Comment by balbirthomas** [comment]
> I recommend asking on the [scipy-user mailing list](https://www.scipy.org/scipylib/mailing-lists.html). Github issues are not productive places for conversational questions like this.

Agreed. I have edited the issue to read like a feature request and will use the scipy-user mailing list for any discussions.

**Comment by j-bowhay** [comment]
While these tools are very useful I think this is too specialised for Scipy and is better served by specialist packages
